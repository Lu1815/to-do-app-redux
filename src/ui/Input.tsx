import { clsx } from 'clsx';
import React, { Dispatch, SetStateAction } from 'react'

interface IInputProps {
    taskName?: string;
    setTaskName: Dispatch<SetStateAction<string>>;
    error: boolean;
    setError: Dispatch<SetStateAction<boolean>>;
}

const Input = ({ taskName, setTaskName, error, setError}: IInputProps) => {
    const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
        setError(false);
        const taskName = event.currentTarget.value;

        setTaskName(taskName);
    }

    return (
        <input 
            id="task-name" 
            type='text' 
            className={clsx('rounded-md p-2 border-gray-200 border-2 outline-none placeholder:text-gray-600 text-black', error ?? 'shake-animation border-red')} 
            placeholder='Task Name...'
            value={taskName}
            onChange={handleChange}
        />
    )
}

export default Input