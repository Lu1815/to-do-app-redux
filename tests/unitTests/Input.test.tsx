import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Input from '../../src/ui/Input';

describe('Input Component', () => {
  test('renders Input with default props', () => {
    render(<Input setTaskName={() => {}} error={false} setError={() => {}} />);
    const inputElement = screen.getByPlaceholderText('Task Name...');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveValue('');
    expect(inputElement).not.toHaveClass('shake-animation');
    expect(inputElement).not.toHaveClass('border-red');
  });

  test('handles input change and calls setTaskName', () => {
    const mockSetTaskName = jest.fn();
    render(<Input setTaskName={mockSetTaskName} error={false} setError={() => {}} />);
    const inputElement = screen.getByPlaceholderText('Task Name...');

    fireEvent.change(inputElement, { target: { value: 'New Task' } });

    expect(mockSetTaskName).toHaveBeenCalledWith('New Task');
  });

  test('handles input change and resets error state', () => {
    const mockSetError = jest.fn();
    render(<Input setTaskName={() => {}} error={true} setError={mockSetError} />);
    const inputElement = screen.getByPlaceholderText('Task Name...');

    fireEvent.change(inputElement, { target: { value: 'New Task' } });

    expect(mockSetError).toHaveBeenCalledWith(false);
  });

  test('matches snapshot', () => {
    const { container } = render(<Input setTaskName={() => {}} error={false} setError={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
