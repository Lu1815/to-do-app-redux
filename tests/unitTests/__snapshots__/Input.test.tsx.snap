// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches snapshot 1`] = `
<div>
  <input
    class="rounded-md p-2 border-gray-200 border-2 outline-none placeholder:text-gray-600 text-black"
    id="task-name"
    placeholder="Task Name..."
    type="text"
    value=""
  />
</div>
`;
